Source/Core/VideoBackends/Metal/MTLUtil.mm

```diff
@@ -0,0 +1,356 @@
+  for (id<MTLDevice> device in devices.Get())
+  {
+    if (device != default_dev)
+      list.push_back(MRCRetain(device));
+  }
+  return list;
+}
+
+void Metal::Util::PopulateBackendInfo(VideoConfig* config)
+{
+  config->backend_info.api_type = APIType::Metal;
+  config->backend_info.bUsesLowerLeftOrigin = false;
+  config->backend_info.bSupportsExclusiveFullscreen = false;
+  config->backend_info.bSupportsDualSourceBlend = true;
+  config->backend_info.bSupportsPrimitiveRestart = true;
+  config->backend_info.bSupportsGeometryShaders = false;
+  config->backend_info.bSupportsComputeShaders = true;
+  config->backend_info.bSupports3DVision = false;
+  config->backend_info.bSupportsEarlyZ = true;
+  config->backend_info.bSupportsBindingLayout = true;
+  config->backend_info.bSupportsBBox = true;
+  config->backend_info.bSupportsGSInstancing = false;
+  config->backend_info.bSupportsPostProcessing = true;
+  config->backend_info.bSupportsPaletteConversion = true;
+  config->backend_info.bSupportsClipControl = true;
+  config->backend_info.bSupportsSSAA = true;
+  config->backend_info.bSupportsFragmentStoresAndAtomics = true;
+  config->backend_info.bSupportsReversedDepthRange = false;
+  config->backend_info.bSupportsLogicOp = false;
+  config->backend_info.bSupportsMultithreading = false;
+  config->backend_info.bSupportsGPUTextureDecoding = true;
+  config->backend_info.bSupportsCopyToVram = true;
+  config->backend_info.bSupportsBitfield = true;
+  config->backend_info.bSupportsDynamicSamplerIndexing = true;
+  config->backend_info.bSupportsFramebufferFetch = false;
+  config->backend_info.bSupportsBackgroundCompiling = true;
+  config->backend_info.bSupportsLargePoints = true;
+  config->backend_info.bSupportsPartialDepthCopies = true;
+  config->backend_info.bSupportsDepthReadback = true;
+  config->backend_info.bSupportsShaderBinaries = false;
+  config->backend_info.bSupportsPipelineCacheData = false;
+  config->backend_info.bSupportsCoarseDerivatives = false;
+  config->backend_info.bSupportsTextureQueryLevels = true;
+  config->backend_info.bSupportsLodBiasInSampler = false;
+  config->backend_info.bSupportsSettingObjectNames = true;
+}
+
+void Metal::Util::PopulateBackendInfoAdapters(VideoConfig* config,
+                                              const std::vector<MRCOwned<id<MTLDevice>>>& adapters)
+{
+  config->backend_info.Adapters.clear();
+  for (id<MTLDevice> adapter : adapters)
+  {
+    config->backend_info.Adapters.push_back([[adapter name] UTF8String]);
+  }
+}
+
+void Metal::Util::PopulateBackendInfoFeatures(VideoConfig* config, id<MTLDevice> device)
+{
+#ifdef TARGET_OS_MAC
+  config->backend_info.bSupportsDepthClamp = true;
+  config->backend_info.bSupportsST3CTextures = true;
+  config->backend_info.bSupportsBPTCTextures = true;
+#else
+  bool supports_mac1 = false;
+  bool supports_apple4 = false;
+  if (@available(iOS 13, *))
+  {
+    supports_mac1 = [device supportsFamily:MTLGPUFamilyMac1];
```

Comment:
You're checking for a Mac GPU family in code that runs on iOS. Is this correct?

Reply:
I thought M1 iPads might support it?  Don't have one to test though.